flang/test/Lower/identical-block-merge-disable.f90

```diff
@@ -0,0 +1,42 @@
+! Test disable identical block merge in the canonicalizer pass in bbc.
+! Temporary fix for issue #1021.
+! RUN: bbc %s -o - | FileCheck %s
+
+MODULE DMUMPS_SOL_LR
+IMPLICIT NONE
+
+TYPE BLR_STRUC_T
+  INTEGER, DIMENSION(:), POINTER  :: PANELS_L 
+  INTEGER, DIMENSION(:), POINTER  :: PANELS_U
+  INTEGER, DIMENSION(:), POINTER :: BEGS_BLR_STATIC
+END TYPE BLR_STRUC_T
+
+TYPE(BLR_STRUC_T), POINTER, DIMENSION(:), SAVE :: BLR_ARRAY
+
+CONTAINS
+
+SUBROUTINE DMUMPS_SOL_FWD_LR_SU( IWHDLR, MTYPE )
+
+  INTEGER, INTENT(IN) :: IWHDLR, MTYPE
+  INTEGER :: NPARTSASS, NB_BLR
+
+  IF (MTYPE.EQ.1) THEN
+    IF ( associated( BLR_ARRAY(IWHDLR)%PANELS_L ) ) THEN
+      NPARTSASS = size( BLR_ARRAY(IWHDLR)%PANELS_L )
+      NB_BLR = size( BLR_ARRAY(IWHDLR)%BEGS_BLR_STATIC ) - 1
+    ENDIF
+  ELSE
+    IF ( associated( BLR_ARRAY(IWHDLR)%PANELS_U ) ) THEN
+      NPARTSASS = size( BLR_ARRAY(IWHDLR)%PANELS_U )
+      NB_BLR = size( BLR_ARRAY(IWHDLR)%BEGS_BLR_STATIC ) - 1
+    ENDIF
+  ENDIF
+
+END SUBROUTINE DMUMPS_SOL_FWD_LR_SU 
+
+END MODULE DMUMPS_SOL_LR
+
+! CHECK: func @_QMdmumps_sol_lrPdmumps_sol_fwd_lr_su
+! CHECK: ^bb[[BLK1:[1-9]]]
+! CHECK: cond_br %{{.*}}, ^bb[[BLKTRUE:[1-9]]], ^bb[[BLKFALSE:[1-9]]]
+! CHECK: ^bb[[BLKTRUE]]: // pred: ^bb[[BLK1]]
```

Comment:
I think we want to error on the side of very restrictive tests before upstreaming (so review suggestions that break the compiler can be quickly caught). Consider just running the bbc output through mlir/utils/generate-test-checks.py and slurping in that as the test.

Reply:
Updated the test with the auto-generated checks.